kornia/augmentation/functional.py

```diff
@@ -327,3 +329,104 @@ def random_rectangle_erase(
     )
     images = erase_rectangles(images, rect_params)
     return images
+
+
+def perspective(input: torch.Tensor,
+                start_points: torch.Tensor,
+                end_points: torch.Tensor,
+                return_transform: bool = False) -> UnionType:
+    r"""Perform perspective transform of the given torch.Tensor or batch of tensors.
+
+    Args:
+        input (torch.Tensor): Tensor to be transformed with shape BxCxHxW.
+        start_points (torch.Tensor): Tensor containing [top-left, top-right, bottom-right,
+        bottom-left] of the orignal image with shape Bx4x2.
+        end_points (torch.Tensor): Tensor containing [top-left, top-right, bottom-right,
+        bottom-left] of the transformed image with shape Bx4x2.
+        return_transform (bool): if ``True`` return the matrix describing the transformation
+        applied to each. Default: False.
+
+    Returns:
+        torch.Tensor:  Perspectively transformed tensor.
+    """
+    if not torch.is_tensor(input):
+        raise TypeError(f"Input type is not a torch.Tensor. Got {type(input)}")
+
+    if not torch.is_tensor(start_points):
+        raise TypeError(f"Input type is not a torch.Tensor. Got {type(start_points)}")
+
+    if not torch.is_tensor(end_points):
+        raise TypeError(f"Input type is not a torch.Tensor. Got {type(end_points)}")
+
+    # compute the homography between the input points
+    transform: torch.Tensor = get_perspective_transform(start_points, end_points)
+
+    # apply the computed transform
+    height, width = input.shape[-2:]
+    img_warped: torch.Tensor = warp_perspective(input, transform, (height, width))
+
+    if return_transform:
+        return img_warped, transform
+
+    return img_warped
+
+
+def random_perspective(input: torch.Tensor,
+                       distortion_scale: float = 0.5,
+                       p: float = 0.5,
+                       return_transform: bool = False) -> UnionType:
+    r"""Performs Perspective transformation of the given torch.Tensor randomly with a given probability.
+
+    Args:
+        input (torch.Tensor): Tensor to be transformed with shape (*, C, H, W).
+        p (float): probability of the image being perspectively transformed. Default value is 0.5
+        distortion_scale(float): it controls the degree of distortion and ranges from 0 to 1. Default value is 0.5.
+        return_transform (bool): if ``True`` return the matrix describing the transformation
+        applied to each. Default: False.
+        input tensor.
+    """
+    if not torch.is_tensor(input):
+        raise TypeError(f"Input type is not a torch.Tensor. Got {type(input)}")
+
+    if len(input.shape) > 4:
+        raise ValueError(f"Input size must have a shape of (*, 3, H, W). Got {input.shape}")
```

Comment:
Are we not accepting inputs of type (*, C, H, W)? if not then why not? What about (C, H, W) inputs? Maybe we can move this check to line we can change this to:

```
x_data: torch.Tensor = input.unsqueeze(0)
x_data = input.view(-1, *input.shape[-3:])
 if len(input.shape) ! = 4:
        raise ValueError(f"Input size must have a shape of (*, C, H, W). Got {input.shape}")
```
In order to accept just one image as input and and arbitrary shapes?

Also notice the 3 must be changed to a C.